### PR TITLE
Tolerate CriticalAddonsOnly, NoExecute, NoSchedule taints

### PR DESCRIPTION
**Pull request by albertocsm**
Proposing adding a series of common taint toleration for the type of workload (DS)

### CODE REVIEW

**Comment by mathieu-benoit** [comment]
Thanks @albertocsm, very important as soon as we taint our nodes, otherwise `kured` won't be Scheduled on tainted nodes... No activities on this since Sep 11th, any chance to get this reviewed and merged?

File: kured-ds.yaml [hunk 1]
```diff
@@ -25,6 +25,12 @@ spec:
       tolerations:
         - key: node-role.kubernetes.io/master
           effect: NoSchedule
+        - key: CriticalAddonsOnly
+          operator: Exists
+        - effect: NoExecute
+          operator: Exists
+        - effect: NoSchedule
```
Comment (bboreham) [inline kured-ds.yaml]: Doesn't this obviate the need for `node-role.kubernetes.io/master` one at the top?

File: kured-ds.yaml [hunk 2]
```diff
@@ -25,6 +25,12 @@ spec:
       tolerations:
         - key: node-role.kubernetes.io/master
           effect: NoSchedule
+        - key: CriticalAddonsOnly
```
Comment (bboreham) [inline kured-ds.yaml]: Why is this needed as well as the other two?  What `effect` does it have?

**Comment by dholbach** [comment]
Just a quick PSA from me:

- we're hanging out in #kured on Slack
- we're planning a contributors meeting soon: here's the [link to the doodle](https://doodle.com/poll/tuwg4w4mxbbknhe3) to find a good time - be sure to pick your timezone before submitting ([preliminary agenda](https://docs.google.com/document/d/1bsHTjHhqaaZ7yJnXF6W8c89UB_yn-OoSZEmDnIP34n8/edit#))

We'd love to see you there.

**Comment by ntimo** [comment]
Could you please also add support for this to the helm chart? Thanks.

**Comment by ckotzbauer** [comment]
@ntimo You can overwrite the tolerations in the chart already, to fit your needs.
